### ISSUE TITLE
"Battery Low" yet 100% shown

### ISSUE THREAD

**Issue by jdshkolnik** [ebaauw/homebridge-hue#280]
Hi,

18 of my accessories are showing low battery in Home while simultaneously showing 100% battery. I'm not sure I trust that 100% but I don't know how to independently verify what's accurate.

![img_0223](https://user-images.githubusercontent.com/2047520/35831225-04976780-0a97-11e8-9484-479c90b2a7bd.JPG)

**Comment by jdshkolnik** [comment]
This seems like it might be related to #223

**Comment by ebaauw** [comment]
Please attach the debug dump file, see https://github.com/ebaauw/homebridge-hue#debug-dump-file.

**Comment by jdshkolnik** [comment]
[homebridge-hue.json.gz](https://github.com/ebaauw/homebridge-hue/files/1697071/homebridge-hue.json.gz)

**Comment by ebaauw** [comment]
I'm sorry, I'm unable to reproduce this and there doesn't seem anything odd in the dump that would explain this.

Some longshots:
- Do you have other homebridge plugins installed?  Does the _Battery Low_ still appear when you have only homebridge-hue installed (I really mean installed, not just configured in config.json)?
- Are you running iOS beta?
- Does the same show on another iOS device?  On another HomeKit app (like Eve)?

**Comment by jdshkolnik** [comment]
Not running iOS beta. I do have other plugins installed. The low battery warning seems to have disappeared. I still can't explain those "get" errors though.

**Comment by ebaauw** [comment]
Which get errors?

**Comment by jdshkolnik** [comment]
Am seeing this error repeated multiple times in journal log:

> Feb 05 16:31:48 raspberrypi homebridge[15326]: [2018-2-5 16:31:48] [Hue] Philips Hue 9: bridge request 387: get /config
> Feb 05 16:31:48 raspberrypi homebridge[15326]: [2018-2-5 16:31:48] [Hue] Philips Hue 9: bridge communication error ECONNREFUSED on 192.***.***.195
> Feb 05 16:31:48 raspberrypi homebridge[15326]: [2018-2-5 16:31:48] [Hue] Philips Hue 9: heartbeat error: Error
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at Request.request [as _callback] (/usr/lib/node_modules/homebridge-hue/lib/HueBridge.js:1006:23)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at self.callback (/usr/lib/node_modules/homebridge-hue/node_modules/request/request.js:186:22)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at emitOne (events.js:116:13)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at Request.emit (events.js:211:7)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at Request.onRequestError (/usr/lib/node_modules/homebridge-hue/node_modules/request/request.js:878:8)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at emitOne (events.js:121:20)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at ClientRequest.emit (events.js:211:7)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at Socket.socketErrorListener (_http_client.js:387:9)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at emitOne (events.js:116:13)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at Socket.emit (events.js:211:7)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at emitErrorNT (internal/streams/destroy.js:64:8)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at _combinedTickCallback (internal/process/next_tick.js:138:11)
> Feb 05 16:31:48 raspberrypi homebridge[15326]: at process._tickCallback (internal/process/next_tick.js:180:9)

**Comment by jdshkolnik** [comment]
I thought I'd hit submit yesterday on that post but obviously not.

**Comment by ebaauw** [comment]
Hmm , I shouldn’t dump the stack trace here - this is more like an exception than an error.

ECONNREFUSED means that the Hue bridge doesn’t accept an incoming http connection (for the REST API call).  Typically this would happen when the webserver on the Hue bridge isn’t listening on port 80.  I’ve only seen this while the Hue bridge is starting up.  If this happens often, it could be that the bridge is crashing and rebooting.

Do you see this on one bridge or on both?  Are all devices that showed _Battery Low_ connected to the same bridge?

**Comment by ebaauw** [comment]
Just noticed: a temperature of 73 degrees?

**Comment by jdshkolnik** [comment]
Seems to be just one bridge but I’m not certain.

What’s strange about the temperature?

**Comment by ebaauw** [comment]
Is seems rather high, or is it in Fahrenheit?

**Comment by jdshkolnik** [comment]
Fahrenheit.

**Comment by MobiusOne91** [comment]
Hi,

in the last weeks (or months) I've seen quite often the battery message for the Motion Sensors. Sometimes I only get that message for 2 and sometimes for 12 devices. I have made screenshot from the Home, Eve and iConnect Apps. In the iConnect App you can see, that the Battery status is unknown. I use your plugin in a separate Homebridge instance, with the following options:
          "sensors": true,
          "excludeSensorTypes": ["ZLLTemperature","CLIPPresence", "Geofence","CLIPGenericStatus","Daylight"],
          "lights": true,
          "nativeHomeKit": false,

Greetings

![home1](https://user-images.githubusercontent.com/30690081/36111031-1d6a1288-1025-11e8-8821-5921a77ce1e5.png)
![home2](https://user-images.githubusercontent.com/30690081/36111037-243eedb8-1025-11e8-907f-b7734460b03c.png)
![eve](https://user-images.githubusercontent.com/30690081/36111047-2be5f598-1025-11e8-9675-6ebbb08be1e7.png)
![iconnect](https://user-images.githubusercontent.com/30690081/36111082-3ccd6dbe-1025-11e8-93da-bda8675f087d.png)

**Comment by ebaauw** [comment]
This is intersting, iConnectHue also shows battery unknown.  I’d really like to see what the sensor resource looks like.  Could you attach the debug dump that homebridge-hue created on startup?  See https://github.com/ebaauw/homebridge-hue#debug-dump-file.

I suspect the bridge reports value `null` for `config.battery`.  Oddly, NodeJS thinks `null <= 25` is true.  That would explain the Status Low Battery.  Updating the Battery value to `null` does nothing, so HomeKit continues to show the previous value.

**Comment by ebaauw** [comment]
As of next version, a `config.battery` of `null` will result in a Status Low Battery _and_ a Battery value of 0%.

Now the real question is, why does the Hue bridge report `config.battery` to be `null`.  It should only do so after startup.  Last time I sniffed the Hue bridge traffic, it configures the motion sensor to report its battery level every 2 hours (for the Hue dimmer switch this is every 5 minutes), so it could remain `null` for up to two hours after startup (longer if the notification fails).

If your bridge restarts spontaneously, you should see communication errors in the homebridge log (like @jdshkolnik above).  Also, you should see a message `set homekit battery level from 100% to null%` on the next successful heartbeat.  You might want to check the [Hue developer forum](https://developers.meethue.com/forum) as to why the bridge restarts.

**Comment by MobiusOne91** [comment]
I added the generated dump-file and added syslog. 

> It should only do so after startup.

You are right. I just updated the Philips Hue firmware, before I made the screenshots. So I think that problem is solved.

When I might ask a different question. Inside syslog I see following status updates:
Bewegungsmelder Badezimmer: add history entry {"time":1518398625,"status":0}

But inside the Eve App the history is shown by the brightness sensor inside the motion category (see screenshot) Is that normal?
[homebridge-hue.json.gz](https://github.com/ebaauw/homebridge-hue/files/1717609/homebridge-hue.json.gz)
[syslog.log](https://github.com/ebaauw/homebridge-hue/files/1717611/syslog.log)
![img_0053](https://user-images.githubusercontent.com/30690081/36117383-a94a3c7e-1039-11e8-9bb2-9a5e0a927ae0.PNG)


.
